src/Core/System/SalesChannel/SalesChannelEntity.php

```diff
@@ -48,516 +48,163 @@ class SalesChannelEntity extends Entity
     use EntityCustomFieldsTrait;
     use EntityIdTrait;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $typeId;
+    protected string $typeId;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $languageId;
+    protected string $languageId;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $currencyId;
+    protected string $currencyId;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $paymentMethodId;
+    protected string $paymentMethodId;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $shippingMethodId;
+    protected string $shippingMethodId;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $countryId;
+    protected string $countryId;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $navigationCategoryId;
+    protected string $navigationCategoryId;
 
-    /**
-     * @var string
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $navigationCategoryVersionId;
+    protected string $navigationCategoryVersionId;
 
-    /**
-     * @var int
-     *
-     * @deprecated tag:v6.7.0 - Will be natively typed
-     */
-    protected $navigationCategoryDepth;
+    protected int $navigationCategoryDepth;
 
     /**
      * @var array<mixed>|null
```

Comment:
```suggestion
     * @var array<string, mixed>|null
```

Reply:
would it not be a break to change a type declaration without deprecating it before and how do you know that the index has to be a string? i do not find any reference for this

Reply:
Annotations are more like browser/static anaylze infos and are not breaking. Would be different, with native typing.

In this case I just happen to know that. SlotConfig always looks like this:
```
                'slotConfig' => [
                    $slotId => [
                        'content' => [
                            'source' => 'static',
                            'value' => $languageCode . ' Override',
                        ],
                        'maybeSomethingDifferent' => 'something of mixed type',
                    ],
                ],
```